doc/conf.py

```diff
@@ -84,6 +84,20 @@
 version = re.sub(r'(\.dev\d+).*?$', r'\1', version)
 numpydoc_xref_param_type = True
 numpydoc_xref_ignore = {'optional', 'type_without_description', 'BadException'}
+# Run docstring validation as part of build process
+numpydoc_validate = True
+numpydoc_validation_checks = {
+    "GL06",  # Unknown section name
+    "GL07",  # Sections in wrong order
+    "GL08",  # Object missing docstring
+    "PR01",  # Missing parameters
+    "PR02",  # Unknown parameters
+    "PR03",  # Incorrect parameter ordering
+    "PR10",  # Missing colon between parameter name and type
+    "RT01",  # No returns section
+    "SA01",  # Missing See Also section
+    "EX01",  # Missing Examples section
+}
```

Comment:
WDYT about allowing `'all'` as an entry, and if there are other entries, treat it as "all except these"? For example to process all but missing examples it would be:
```
numpydoc_validation_checks = {'all', 'EX01', 'SA01'}
```
Some variant of this is probably what I would use in most projects. Basically it would be nice to provide a convenient way to turn it into a blocklist rather than allowlist. Or maybe it should be `'not'` or `'reversed'` or something instead of `'all'`, not sure.

Reply:
I agree, but I don't know what the best way is. I thought the set approach might be good because then you can subtract sets (e.g. `set(ERROR_MSGS.keys()) - set(('GL01', 'EX01'))`) to achieve fine-grained control in a clear (albeit ugly) way.

Reply:
Yes that works but then people have to do more work like import `ERROR_MSGS` and/or know where they are in the namespace. I guess we could put it in `numpdoc` directly to ameliorate that issue.

To me providing `{'not', 'GL01', 'EX01'}` is clear enough, convenient enough from the user end, and easy enough from a code end just to add as a shortcut. But if you don't see the value I can live without it

Reply:
Sounds good - I'll give that a try, along with adding better documentation for the error codes.